test cases/common/262 required keyword in has functions/meson.build

```diff
@@ -0,0 +1,55 @@
+project('required keyword in has functions', 'c')
+
+cc = meson.get_compiler('c')
+opt = get_option('opt')
+
+cc.has_function('printf', prefix : '#include<stdio.h>', required : true)
```

Comment:
Does it make sense to add tests for the other `has` functions which already support required? `has_symbol_header` for instance.

Reply:
I thought those were already tested elsewhere, but it seems not. Added now.